web/server/codechecker_server/api/report_server.py

```diff
@@ -1549,14 +1549,16 @@ def getReportDetails(self, reportId):
         with DBSession(self.__Session) as session:
             return get_report_details(session, [reportId])[reportId]
 
-    def _setReviewStatus(self, session, report_id, status, message, date=None):
+    def _setReviewStatus(self, session, report_hash, status,
+                         message, date=None):
         """
         This function sets the review status of the given report. This is the
         implementation of changeReviewStatus(), but it is also extended with
         a session parameter which represents a database transaction. This is
         needed because during storage a specific session object has to be used.
         """
-        report = session.query(Report).get(report_id)
+        report = session.query(Report).filter(Report.bug_id == report_hash)\
+            .all()[0]
```

Comment:
Review statuses are connected to report hashes. Previously we fetched the report from the database because the report id was given to this function and we needed the report hash for this report. Now we have the report hash in the parameter so there is no need to fetch the report here.
You can remove this line the if condition and the else branch and indent the code like this in the if branch:
```suggestion
        review_status = session.query(ReviewStatus).get(report_hash)
        ...
```